raft/raftpb/raft.proto

```diff
@@ -64,6 +64,7 @@ message Message {
 	optional Snapshot    snapshot    = 9  [(gogoproto.nullable) = false];
 	optional bool        reject      = 10 [(gogoproto.nullable) = false];
 	optional uint64      rejectHint  = 11 [(gogoproto.nullable) = false];
+	optional bytes       context     = 12;
```

Comment:
@siddontang what you suggested makes sense for this single case, but I hope it can be more generic which can be used for more cases, such as ReadIndexMsg, it still use Entries of Message to store some user context, i wanna store that context into this field too.

thanks